src/server/stream_family.cc

```diff
@@ -1640,6 +1850,54 @@ void StreamFamily::XRead(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void XReadGeneric(CmdArgList args, bool read_group, ConnectionContext* cntx) {
+  auto opts = ParseReadArgsOrReply(args, read_group, cntx);
+  if (!opts) {
+    return;
+  }
+
+  vector<vector<GroupConsumerPair>> res_pairs(shard_set->size());
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    if (!opts->read_group) {
```

Comment:
I do not understand this condition. When does it happen that `opts->read_group` is false? 
what's the expected behavior in that case?

Reply:
In case of `xread` group command.

Reply:
I prefer that this logic will be done earlier in the flow, i.e. it seems you do not need to schedule this hop at all for XREAD. Instead of checking inside shard callback, 
you can probably run it like this after the schedule call.

```
if (opts->read_group) {
  cntx->transaction->Execute(std::move(cb), false);
}
```

Reply:
sounds good, thanks